GVFS/GVFS.Common/ProductUpgraderPlatformStrategyFactory.cs

```diff
@@ -0,0 +1,23 @@
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using System.Runtime.InteropServices;
+
+namespace GVFS.Common
+{
+    public static class ProductUpgraderPlatformStrategyFactory
+    {
+        public static ProductUpgraderPlatformStrategy Create(
+            PhysicalFileSystem fileSystem,
+            ITracer tracer)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return new MacProductUpgraderPlatformStrategy(fileSystem, tracer);
+            }
+            else

```

Comment:
Perhaps be specific here and have the `else` condition throw a `NotSupportedException` or something? That way the Linux platform can slot in easilyi.

Reply:
I liked this suggestion - but moved the "Creation" logic into the Platform classes, so that should help with the Linux platform.